src/markov/sean.clj

```diff
@@ -0,0 +1,98 @@
+(ns markov.sean
+  (:gen-class))
+
+;; Generating a markov dictionary from a string of text
+;; e.g. { :first-words ["The" "An"] :last-words ["dog"] :dict { "The" ["quick" "large"] "fox" ["jumped"] ... } }
+
+(defn zip-pairs
+  "From a list, generates a list of sequential pairs ([1 2 3] becomes [[1 2] [2 3]])"
+  [list]
+  (map vector list (next list)))
+
+(defn build-dict
+  "Builds the graph of words in a markov dictionary, given an ordered list of words"
+  [words]
+  (reduce (fn [dict, current-pair]
+            (let [current-word (current-pair 0)
+                  next-word (current-pair 1)
+                  words-after-current (get dict current-word)]
+              (if words-after-current
+                (update dict current-word conj next-word) 
+                (merge dict {current-word [next-word]}))))
+          {} (zip-pairs words)))
+
+(defn split-into-words
+  [text]
+  (clojure.string/split text #" "))
+
+(defn get-first-words
+  "From a list of sentences, returns a list of the first words"
+  [sentences]
+  (map (fn [sentence]
+         (first (split-into-words sentence))) sentences))
+
+(defn get-last-words
+  "From a list of sentences, returns a list of the last words"
+  [sentences]
+  (map (fn [sentence]
+         (last (split-into-words sentence))) sentences))
+
+(defn markov-generate-dict
+  "Builds a markov dict for some text, including a list of first and last words."
+  [text]
+  (let [sentences (clojure.string/split text #"\. ") ; TODO: handle '?'
```

Comment:
Up in your `ns` declaration you can use

```clojure
(ns foo.bar
 (:require [clojure.string :as str])
```

or

```clojure
(ns foo.bar
 (:require [clojure.string :refer [split]])
```

to pull things in and make them a bit less verbose as you call them.